extmod/modtls_mbedtls.c

```diff
@@ -100,11 +100,22 @@ static void mbedtls_debug(void *ctx, int level, const char *file, int line, cons
 }
 #endif
 
-#if defined(MBEDTLS_PEM_PARSE_C)
-static int mbedtls_is_pem(const byte *data, size_t len) {
-    return (len >= 10) && (strstr((const char *)data, "-----BEGIN") != NULL);
+// Given a string-like object holding PEM or DER formatted ASN.1 data, return a
+// pointer to its buffer and the correct length for mbedTLS APIs.
+//
+// (mbedTLS >= 3.5 rejects DER formatted data with trailing bytes within keylen,
+// but PEM must include a terminating NUL byte in the keylen...)
+static const byte *asn1_get_data(mp_obj_t obj, size_t *out_len) {
```

Comment:
Considering the mbedtls API uses `const unsigned char *` as the data type for key/cert, perhaps it's worth using that here as the return type, so it matches exactly?  (I know it doesn't really matter, but it may aid in self-documenting the code?)